src/components/ingestion/FileIngestionEngine.js

```diff
@@ -0,0 +1,33 @@
+import fs from 'fs'
+import AbstractIngestionEngine from '%src/components/ingestion/AbstractIngestionEngine'
+
+/**
+ * The FileIngestionEngine handles processing a video file. It is a concrete
+ * implementation of AbstractIngestionEngine, which reads in a file,
+ * converts it into an MPEG-TS stream represented as a series of Payloads,
+ * and then pipes that to Kafka.
+ *
+ * @param  {string}  path The path of the file to be ingested
+ */
+class FileIngestionEngine extends AbstractIngestionEngine {
+	constructor(path) {
+		if (!path) {
+			throw new Error('path required') // TODO: should probably be FileInstantiationError
```

Comment:
Just noting since there's a TODO here -- should it be or not?  If it should, should we just do that here?  If not, should we take out the TODO?

From what I can tell, the benefit of custom errors is:

1. It lets you specify extra metadata (probably not relevant in this case)
2. It allows for more specific try / catch blocks in general.

It might also be good fodder for `CONTRIBUTING.md` if we land on a project philosophy around errors.

Note that we *would not have to add this error to base-errors*.  That package is for errors that are thrown by `base-classes` or `base-interfaces`.  The point being: we could have a custom `error` in this repository directly (which is just a loser barrier to entry.

I'm comfortable moving in that direction now -- including an update to CONTRIBUTING.md in this PR with a note akin to "Never use a vanilla `Error`, always try to use an existing custom error type.")  Unless we hate that!  In which case we should probably just document what we want.

I do think that the custom error should probably be a slightly more generic `InstantiationError` rather than `FileIngestionInstantiationError` (unless `FileInstantiationError` is a thing and you meant it!).

Reply:
I was looking to the errors in AsbtractIngestionEngine for guidance. Happy to do whatever, but agree adding it to the base package and defining a new error for every circumstance (as opposed to something slightly more generic) seemed excessive, which was why it was a TODO.

Reply:
I was googling "does anybody care about custom errors in Node" and got basically zero results so I feel lost and afraid.

Reply:
Based on the conversation in Slack, I did not define a custom error. Should I have?

Reply:
That was the right call!